plugins/chat/spec/system/transcript_spec.rb

```diff
@@ -0,0 +1,247 @@
+
+    it "quotes a single chat message into a topic " do
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
+
+      clip_text = copy_messages_to_clipboard(chat_message_5)
+
+      # post transcript in topic
+      topic_page.visit_topic_and_open_composer(topic)
+      topic_page.fill_in_composer("This is a new post!\n\n" + clip_text)
+      within ".d-editor-preview" do
+        expect(page).to have_css(".chat-transcript")
+      end
+      topic_page.send_reply
+      expect(page).to have_content("This is a new post!")
+      within topic_page.post_by_number(topic.posts.reload.last.post_number) do
+        expect(page).to have_css(".chat-transcript")
+      end
+    end
+
+    it "quotes multiple chat messages into a topic" do
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
+
+      messages = [chat_message_5, chat_message_4, chat_message_3, chat_message_2]
+      clip_text = copy_messages_to_clipboard(messages)
+
+      # post transcript in topic
+      topic_page.visit_topic_and_open_composer(topic)
+      topic_page.fill_in_composer("This is a new post!\n\n" + clip_text)
+      within ".d-editor-preview" do
+        expect(page).to have_css(".chat-transcript", count: 4)
+      end
+      expect(page).to have_content("Originally sent in #{chat_channel_1.name}")
+      topic_page.send_reply
+      expect(page).to have_content("This is a new post!")
+      within topic_page.post_by_number(topic.posts.reload.last.post_number) do
+        expect(page).to have_css(".chat-transcript", count: 4)
+      end
+    end
+
+    it "does not error in preview when quoting a chat message with a onebox" do
+      Oneboxer.stubs(:preview).returns(
+        "<aside class=\"onebox\"><article class=\"onebox-body\"><h3><a href=\"http://www.example.com/article.html\" tabindex=\"-1\">An interesting article</a></h3></article></aside>",
+      )
+      chat_message_3.update!(message: "http://www.example.com/has-title.html")
+      chat_message_3.rebake!
+
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
+
+      clip_text = copy_messages_to_clipboard(chat_message_3)
+
+      # post transcript in topic
+      topic_page.visit_topic_and_open_composer(topic)
+      topic_page.fill_in_composer(clip_text)
+
+      within ".chat-transcript-messages" do
+        expect(page).to have_content("An interesting article")
+      end
+    end
+
+    it "quotes a single chat message into another chat message " do
+      chat_page.visit_channel(chat_channel_1)
+      expect(chat_channel_page).to have_no_loading_skeleton
+      expect(chat_channel_page).to have_content(chat_message_5.message)
```

Comment:
Not needed given we call `copy_messages_to_clipboard` after which will fail if the message is not here.